### ISSUE TITLE
spaCy Language.__call__ returns Unknown instead of Doc

### ISSUE THREAD

**Issue by sethidden** [microsoft/pyright#2468]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
When a variable with the type `Language` from `spacy.language` is called (`__call__`), pyright thinks the returned type is Unknown instead of Doc - https://github.com/explosion/spaCy/blob/master/spacy/language.py#L984

**To Reproduce**
```python
import spacy

# those imports are not necessary for repro. Just so you know where the affected types come from
from spacy.language import Language
from spacy.tokens import Doc

language = spacy.load("pl_core_news_lg") # if you wanted to actually run this code you'd need to download this dataset, but for purposes of repro running the code is not necessary
reveal_type(language) # type of "language" is "Language" -- this is ok
doc = language("Some text")
reveal_type(doc) # type of "doc" is "Unknown" -- this is not ok - different than what's in https://github.com/explosion/spaCy/blob/master/spacy/language.py#L984

```

**Expected behavior**
spacCy's `Language.__call__()` should return spaCy's `Doc` type instead of `Unknown`.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
![image](https://user-images.githubusercontent.com/5359825/138230982-5fe923ae-5255-4e48-9127-4f1198a7b210.png)

**VS Code extension or command-line**
I use neovim's native LSP client, but in that case I assume the answer is command-line. Version is `1.1.180`

**Additional context**
The typedefs for the Doc class are here:
https://github.com/explosion/spaCy/blob/master/spacy/tokens/doc.pyi
